### PR TITLE
Improve argument about test selection for a feature

### PR DESCRIPTION
**Pull request by pahmann**
As part of the process audit on part 6 clause 1041 an argumentation which test levels are selected for a feature shall be explained. Additionally, the PR adds quality goals which were missing for these tests.

Related issue: #1085

### CODE REVIEW

File: docs/platform_management_plan/software_verification.rst [hunk 1]
```diff
@@ -103,6 +103,20 @@ The following types of integrations are applicable:
...
+of the software element under test. This shall ensure proper testing from unit level to feature integration level.
+Specific recommendations for each test level are provided in the following sections of this verification plan.
+
+The following examples serve as illustration and inspiration for test selection:
+Resource Usage Evaluation is selected when a vulnerability is identified in the software element or when data
+corruption may occur. Interface Testing is applied when an external API is part of a potential attack path or
+when interface problems could cause safety failures. Boundary Value Analysis is used when buffer overflow is possible
+or when incorrect input values could compromise safety.
+
+These examples shall illustrate how test selection is driven by specific risks to ensure that threats and hazards
```
Comment (masc2023) [inline docs/platform_management_plan/software_verification.rst]: In my opinion, that is not according the standard way, identified risks are managed in a way, that either results in removing them or reducing them or add additional elements, which finally have requirements, etc. they are tested.

Comment (pahmann) [reply docs/platform_management_plan/software_verification.rst]: Right. I will try to rephrase it or remove the part.

File: docs/platform_management_plan/software_verification.rst [hunk 2]
```diff
@@ -307,6 +321,14 @@ to be reached with every contribution.
...
      - 0%
      - 0%
+   * - 8
+     - Open rule violations for static code analysis
+     - no major violation based on tool rating
+     - 0
+   * - 9
+     - Compiler warnings
+     - 0
+     - 0
```
Comment (PandaeDo) [inline docs/platform_management_plan/software_verification.rst]: The table is a little bit confusing because the target is 100% and also 0. So it might be more readable when you rephrase it or split it.

Comment (pahmann) [reply docs/platform_management_plan/software_verification.rst]: Updated.

File: docs/platform_management_plan/software_verification.rst (hunk 1 above)
Comment (pahmann) [reply docs/platform_management_plan/software_verification.rst]: Updated.

File: docs/platform_management_plan/software_verification.rst [hunk 3]
```diff
@@ -304,9 +315,17 @@ to be reached with every contribution.
...
-     - 0%
+     - Relative amount of passed tests
+     - 100%
+     - 100%
+   * - 8
+     - Compliance with rules for static code analysis
+     - only minor violation based on tool rating
+     - 100%
+   * - 9
+     - Freedom from compiler warnings & errors
```
Comment (masc2023) [inline docs/platform_management_plan/software_verification.rst:326]: Is that not just #compiler warnings/#errors -> 0/0

File: docs/platform_management_plan/software_verification.rst [hunk 4]
```diff
@@ -304,9 +315,17 @@ to be reached with every contribution.
...
    * - 7
-     - Relative amount of failed tests
-     - 0%
-     - 0%
+     - Relative amount of passed tests
+     - 100%
+     - 100%
+   * - 8
+     - Compliance with rules for static code analysis
+     - only minor violation based on tool rating
```
Comment (masc2023) [inline docs/platform_management_plan/software_verification.rst:323]: minor is not measurable?

**Review by masc2023 — APPROVED** [review]
See comments on table, would prefer concrete measurable values

File: docs/platform_management_plan/software_verification.rst (hunk 4 above)
Comment (pahmann) [reply docs/platform_management_plan/software_verification.rst:323]: Right, but I was not feeling confident to write any number or something. I hope to get a clearer picture when I look at some reports and discuss with committers.
